zenml/core/pipelines/nlp_pipeline.py

```diff
@@ -0,0 +1,207 @@
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""ZenML NLP Pipeline Prototype."""
+
+import os
+from typing import Dict, Text, Any, List, Optional, Union
+
+import tensorflow as tf
+from tfx.components.schema_gen.component import SchemaGen
+from tfx.components.statistics_gen.component import StatisticsGen
+from tfx.components.trainer.component import Trainer
+from tfx.proto import trainer_pb2
+
+from zenml.core.backends.training.training_base_backend import \
+    TrainingBaseBackend
+from zenml.core.components.data_gen.component import DataGen
+from zenml.core.components.split_gen.component import SplitGen
+from zenml.core.components.tokenizer.component import Tokenizer
+from zenml.core.pipelines.base_pipeline import BasePipeline
+from zenml.core.standards import standard_keys as keys
+from zenml.core.steps.split.base_split_step import BaseSplit
+from zenml.core.steps.tokenizer.base_tokenizer import BaseTokenizer
+from zenml.core.steps.trainer.base_trainer import BaseTrainerStep
+from zenml.utils import constants
+from zenml.utils.enums import GDPComponent
+from zenml.utils.enums import PipelineStatusTypes
+
+
+class NLPPipeline(BasePipeline):
+    PIPELINE_TYPE = "nlp"
+
+    def __call__(self, sequence: Union[Text, List[Text]]):
+        """Call operator for local inference method"""
+
+        if not self.get_status() == PipelineStatusTypes.Succeeded.name:
+            print("Please run the pipeline first before running inference!")
+            return
+
+        trainer_step = self.steps_dict[keys.NLPSteps.TRAINER]
+
+        # e.g. HuggingFace has special APIs for model loading,
+        # so we fall back to a trainer step class method
+        model_uri = os.path.join(self.get_model_uri(), "serving_model_dir")
+        model = trainer_step.load_model(model_uri)
+
+        tokenizer_step = self.steps_dict[keys.NLPSteps.TOKENIZER]
+
+        tokenizer_step.load_vocab(self.get_tokenizer_uri())
+
+        encoded = tokenizer_step.encode(sequence=sequence,
+                                        output_format="tf_tensors")
+
+        transformed_bert_features = {k: tf.reshape(v, (1, -1)) for k, v in
+                                     encoded.items()}
+
+        prediction = model(input_ids=transformed_bert_features, training=False)
+
+        formatted = [
+            {"label": model.config.id2label[item.argmax()],
+             "score": item.max().item()}
+            for item in tf.math.sigmoid(prediction.logits).numpy()
+        ]
+
+        print(formatted)
```

Comment:
Should we take this out? Things like these should be logged with a proper logging level and formatting

Reply:
yeah I think the idea was nice, but it is getting hampered a little bit by the fact that you are not only getting the result but also some other logs about metadata connections, which clutter the output. We can take it out for the time being

Reply:
Refactored into `NLPPipeline.predict_sentence` in 7a93c53